core/src/main/java/org/elasticsearch/search/sort/GeoDistanceSortBuilder.java

```diff
@@ -193,14 +294,192 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
+        String nestedPath = null;
+
+        boolean coerce = GeoDistanceSortBuilder.DEFAULT_COERCE;
+        boolean ignoreMalformed = GeoDistanceSortBuilder.DEFAULT_IGNORE_MALFORMED;
+
+        XContentParser.Token token;
+        String currentName = parser.currentName();
+        while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
+            if (token == XContentParser.Token.FIELD_NAME) {
+                currentName = parser.currentName();
+            } else if (token == XContentParser.Token.START_ARRAY) {
+                parseGeoPoints(parser, geoPoints);
+
+                fieldName = currentName;
+            } else if (token == XContentParser.Token.START_OBJECT) {
+                // the json in the format of -> field : { lat : 30, lon : 12 }
+                if ("nested_filter".equals(currentName) || "nestedFilter".equals(currentName)) {
+                    nestedFilter = context.parseInnerQueryBuilder();
+                } else {
+                    fieldName = currentName;
+                    GeoPoint point = new GeoPoint();
+                    GeoUtils.parseGeoPoint(parser, point);
+                    geoPoints.add(point);
+                }
+            } else if (token.isValue()) {
+                if ("reverse".equals(currentName)) {
+                    reverse = parser.booleanValue();
+                } else if ("order".equals(currentName)) {
+                    reverse = "desc".equals(parser.text());
+                } else if ("unit".equals(currentName)) {
+                    unit = DistanceUnit.fromString(parser.text());
+                } else if ("distance_type".equals(currentName) || "distanceType".equals(currentName)) {
+                    geoDistance = GeoDistance.fromString(parser.text());
+                } else if ("coerce".equals(currentName) || "normalize".equals(currentName)) {
+                    coerce = parser.booleanValue();
+                    if (coerce == true) {
+                        ignoreMalformed = true;
+                    }
+                } else if ("ignore_malformed".equals(currentName)) {
+                    boolean ignore_malformed_value = parser.booleanValue();
+                    if (coerce == false) {
+                        ignoreMalformed = ignore_malformed_value;
+                    }
+                } else if ("sort_mode".equals(currentName) || "sortMode".equals(currentName) || "mode".equals(currentName)) {
+                    sortMode = MultiValueMode.fromString(parser.text());
+                } else if ("nested_path".equals(currentName) || "nestedPath".equals(currentName)) {
+                    nestedPath = parser.text();
+                } else {
+                    GeoPoint point = new GeoPoint();
+                    point.resetFromString(parser.text());
+                    geoPoints.add(point);
+                    fieldName = currentName;
+                }
+            }
+        }
+
+        if (sortMode == null) {
```

Comment:
This cannot happen anymore when setting the default, but since this was the behaviour in the "old" style parser, I wonder if we need to keep it?